app/models/user.php

```diff
@@ -2,7 +2,57 @@
 
 class User extends AppModel
 {
-    public function getUserDetail($id)
+    const MIN_STRING_LENGTH = 1;
+    const MAX_STRING_LENGTH = 80;
+
+    public $validation = array(
+            'new_username' => array(
+                'length' => array(
+                    'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH
+                ),
+                'valid' => array(
+                    'validate_username'
+                ),
+            ),
+            'new_first_name' => array(
+                'length' => array(
+                    'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH
+                ),
+                'valid' => array(
+                    'validate_name'
+                ),
+            ),
+            'new_last_name' => array(
+                'length' => array(
+                    'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH
+                ),
+                'valid' => array(
+                    'validate_name'
+                ),
+            ),
+            'new_email' => array(
+                'length' => array(
+                    'validate_between', self::MIN_STRING_LENGTH, self::MAX_STRING_LENGTH
+                ),
+            ),
+
+        );
+
+    public function __construct($data)
+    {
+        if (!is_array($data)) {
+            securedPage();
+            $this->username         = $data;
+            $this->user_id          = $this->getUserId($this->username);
+            $this->user_details     = objectToArray($this->getUserDetail($this->user_id));
+            $this->followed_threads = $this->followedThreads();
```

Comment:
We usually don't put processes like this in the constructor method especially database query. constructor's duty is mostly for declaration and initialisation of object properties.

maybe you can just put it in a separate function and call it? like:
```php
User::getInfoByUsername($user_name);
```